build error when compiling `swc_ecma_ast` and `swc_ecma_minifier` with `swc_ast_unknown` flag

### Describe the bug

Receive the following error when compiling `swc_ecma_ast` and `swc_ecma_minifier` with `swc_ast_unknown` flag.

```
error[E0004]: non-exhaustive patterns: `&_` not covered
   --> /Users/alpheus/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/swc_ecma_minifier-55.0.0/src/compress/optimize/util.rs:523:37
    |
523 |                     let sym = match &e.prop {
    |                                     ^^^^^^^ pattern `&_` not covered
    |
note: `swc_ecma_ast::MemberProp` defined here
   --> /Users/alpheus/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/swc_ecma_ast-25.0.0/src/expr.rs:894:1
    |
894 | pub enum MemberProp {
    | ^^^^^^^^^^^^^^^^^^^
    = note: the matched value is of type `&swc_ecma_ast::MemberProp`
    = note: `swc_ecma_ast::MemberProp` is marked as non-exhaustive, so a wildcard `_` is necessary to match exhaustively
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
530 ~                         MemberProp::PrivateName(_) => break 'a,
531 ~                         &_ => todo!(),
    |

For more information about this error, try `rustc --explain E0004`.
error: could not compile `swc_ecma_minifier` (lib) due to 1 previous error
```

### Input code

```typescript

```

### Config

```json

```

### Link to the code that reproduces this issue

https://github.com/alpheusmtx/swc_ecma_ast_swc_ecma_minifier_swc_ast_unknown

### SWC Info output

_No response_

### Expected behavior

It is expected to be compatible between different internal dependencies.

### Actual behavior

_No response_

### Version

68.0.2

### Additional context

_No response_

--- a/.changeset/handle-unknown-minifier-member-props.md
+++ b/.changeset/handle-unknown-minifier-member-props.md
@@ -1,0 +1,6 @@
+---
+swc_core: patch
+swc_ecma_minifier: patch
+---
+
+fix(es/minifier): Handle unknown member props in hoisted property optimization

--- a/crates/swc_ecma_minifier/src/compress/optimize/util.rs
+++ b/crates/swc_ecma_minifier/src/compress/optimize/util.rs
@@ -528,6 +528,8 @@ impl VisitMut for Finalizer<'_> {
                             _ => break 'a,
                         },
                         MemberProp::PrivateName(_) => break 'a,
+                        #[cfg(swc_ast_unknown)]
+                        _ => break 'a,
                     };
 
                     if let Some(ident) = self.hoisted_props.get(&(obj.to_id(), sym.clone())) {

